### PR TITLE
Make s3io.Reader an io.Seeker

### PR DESCRIPTION
**Pull request by mattnibs**
This functionality is needed for the zar find command to work.

Add system test for zar index and zar find commands.

Part of #972 & #973

### CODE REVIEW

File: pkg/s3io/reader.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	wab := aws.NewWriteAtBuffer(p)
+	bytesDownloaded, err := r.downloader.Download(wab, getObj)
+	if err != nil {
+		return 0, err
+	}
+
+	buf := wab.Bytes()
+	if len(buf) > n {
```
Comment (nwt) [inline pkg/s3io/reader.go:85]: Nit: I think `p == wab.Bytes()` would be a clearer condition.

File: pkg/s3io/reader.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	case io.SeekStart:
+	case io.SeekCurrent:
+		offset += r.offset
+	case io.SeekEnd:
+		offset += r.size
+	default:
+		return 0, fmt.Errorf("%d: unknown whence value", whence)
+	}
+	if offset < 0 {
+		return 0, errInvalidOffset
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply pkg/s3io/reader.go:53]: Awesome

File: pkg/s3io/reader.go (hunk 1 above)
Comment (mattnibs) [reply pkg/s3io/reader.go:80]: You can only compare []byte to `nil`.

File: pkg/s3io/reader.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	if err != nil {
+		return 0, err
+	}
+
+	buf := wab.Bytes()
+	if len(buf) > len(p) {
+		// backing buffer reassigned, copy over some of the data
+		copy(p, buf)
+		bytesDownloaded = int64(len(p))
+	}
```
Comment (nwt) [inline pkg/s3io/reader.go:84]: Nit: The io.WriterAt documentation says WriteAt should return an error on a short write, but using aws.WriteAtBuffer like this effectively ignores a short write. Maybe use this instead.
```go
type writeAtBuf []byte

func (w writeAtBuf) WriteAt(p []byte, off int64) (int, error) {
	n := copy(w[off:], p)
	if n < len(p) {
		return n, errors.New("s3io: short write")
	}
	return n, nil
}
```

Comment (mattnibs) [reply pkg/s3io/reader.go:84]: #attentionToDetail

**Pull request merged**